File: users-service/src/main/java/net/plshark/users/auth/service/NoneAlgorithmBuilder.java
package net.plshark.users.auth.service;

import com.auth0.jwt.algorithms.Algorithm;
import net.plshark.users.auth.AuthProperties;
import org.springframework.stereotype.Component;

/**
 * Builder for the NONE algorithm
 */
@Component
public class NoneAlgorithmBuilder implements AlgorithmBuilder {

    public static final String NONE = "none";

    @Override
    public Algorithm build(AuthProperties authProperties) {
        Algorithm algorithm = null;
        String name = authProperties.getAlgorithm();
        if (NONE.equals(name))
            algorithm = Algorithm.none();
        return algorithm;
    }
}

File: users-service/src/main/kotlin/net/plshark/users/auth/service/NoneAlgorithmBuilder.kt
package net.plshark.users.auth.service

import com.auth0.jwt.algorithms.Algorithm
import net.plshark.users.auth.AuthProperties
import org.springframework.stereotype.Component

/**
 * Builder for the NONE algorithm
 */
@Component
class NoneAlgorithmBuilder : AlgorithmBuilder {

    override fun build(authProperties: AuthProperties): Algorithm? {
        var algorithm: Algorithm? = null
        val name: String = authProperties.algorithm
        if (NONE == name)
            algorithm = Algorithm.none()
        return algorithm
    }

    companion object {
        const val NONE = "none"
    }
}
